### DISCUSSION TITLE
File grid adding file meta data

### DISCUSSION THREAD

**Discussion by blue-dreamer** [ExpressionEngine/ExpressionEngine#2855] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
When adding files to the File grid field there's no way to add file meta data (Title, Description, Credit, Location). This forces the author to go into Files, locate their uploaded files and add the data, that's if they have permission to access the Files CP area.

**Describe the solution you'd like**
Two possible  solutions:

1. Open a modal for each file where meta data can be added, in the case of multiple uploads the next modal would open after the previous had been saved.

2. Add a link with each file so the author could access and edit file meta data. Probably a good enhancement generally.

**Describe alternatives you've considered**
No alternatives exist

**Teachability, Documentation, Adoption, Migration Strategy**
File meta data is already there so it's just a change in process.

- [ ] I am capable and would like to work on implementation of this feature if it is considered.

**Comment by jmathias** [comment]
Thanks Rob, We actually have UI for this functionality. Just have not implemented it yet.

**Comment by blue-dreamer** [comment]
That's perfect, thanks for the update James :)

**Comment by intoeetive** [comment]
@TomJaeger should we consider allowing editing file info from within the File field same way that we allow editing related entry from Relationship field?

Also this has been requested for stand-alone File fields https://github.com/ExpressionEngine/ExpressionEngine/discussions/3151

**Comment by chunkitdesign** [comment]
I discussed it with @TomJaeger and he agreed with what I suggested

**Reply by chunkitdesign** [reply]
May you reopen this discussion?

**Comment by intoeetive** [comment]
This will be implemented in https://github.com/ExpressionEngine/ExpressionEngine/pull/3424 (currently in progress)

**Reply by intoeetive** [reply]
About to be released with 7.4.0
